### PR TITLE
Invalid regexp literals should throw syntax error in ES11

### PR DESCRIPTION
**Pull request by rerobika**
JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: tests/test262-esnext-excludelist.xml [hunk 1]
```diff
@@ -668,229 +667,6 @@
   <!-- END - Uncategorized failing tests -->
 
   <!-- regression caused by: Invalid regexp patterns should not throw syntax error during parsing (#4038) -->
```
Comment (galpeter) [inline tests/test262-esnext-excludelist.xml:669]: This start comment can be removed and also the end comment for this block.

Comment (rerobika) [reply tests/test262-esnext-excludelist.xml:669]: Thanks!

File: tests/jerry/fail/regression-test-issue-2058.js [hunk 2]
```diff
@@ -12,9 +12,4 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-try {
-  /?:/
-  assert(false);
-} catch (e) {
-  assert(e instanceof SyntaxError);
-}
+/?:/
```
Comment (galpeter) [inline tests/jerry/fail/regression-test-issue-2058.js:15]: Would it make sense to have these tests in an eval call and have correct error type check?

Comment (rerobika) [reply tests/jerry/fail/regression-test-issue-2058.js:15]: Sure!

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**